src/utils/dom/selectNodesWithEq.js

```diff
@@ -24,34 +23,15 @@ const EQ_PATTERN = /:eq\((\d+)\)/g;
 const cleanUp = str => str.replace(SIBLING_PATTERN, "").trim();
 const isNotEqSelector = str => str.indexOf(EQ_START) === -1;
 
-const createPair = match => {
-  const first = match.charAt(0);
-  const second = match.charAt(1);
-  const third = match.charAt(2);
-  const result = { key: match };
-
-  if (second === "-") {
-    result.val = `${first}${second}\\3${third} `;
-  } else {
-    result.val = `${first}\\3${second} `;
-  }
-
-  return result;
-};
-
 export const escapeDigitsInSelector = selector => {
-  const matches = selector.match(DIGIT_IN_SELECTOR_PATTERN);
-
-  if (isNonEmptyArray(matches)) {
-    const pairs = matches.map(createPair);
-
-    return pairs.reduce(
-      (acc, pair) => acc.replace(pair.key, pair.val),
-      selector
-    );
-  }
-
-  return selector;
+  const replace = value =>
+    value.replace(DIGIT_IN_SELECTOR_PATTERN, `$1$2\\3$3 $4`);
+
+  return selector
+    .split(" ")
```

Comment:
Why split on space and then join?

Reply:
@jonsnyder I've tried without `split(" ")` and the capture groups would replace ONLY the first element, that's I why I decided to split modify and the re-combine.

I am not sure if there is a better way or more elegant way.

Reply:
I see.  It's because you have (.*) at the end of the pattern.  Regex's will match the longest string possible.  If you removed that, I think it would work. i.e. ```value.replace(/(#|\.)(-)?(\d{1})/g, `$1$2\\3$3 `);```

Reply:
@jonsnyder thanks for your help, I have removed the capture all group and it works as expected.